spicy/toolchain/src/compiler/parser/driver.cc

```diff
@@ -133,3 +133,18 @@ void Driver::processPreprocessorLine(const std::string_view& directive, const st
         case hilti::util::SourceCodePreprocessor::State::Skip: _scanner->setIgnoreMode(true); break;
     }
 }
+
+void Driver::docSummary(const std::string& s) { _doc.addSummary(s); }
+void Driver::docText(const std::string& s) { _doc.addText(s); }
+void Driver::docField(const std::string& s) {
+    // TODO
+}
+
+void Driver::docDeclaration(Declaration* d) {
+    if ( _doc )
+        d->setDocumentation(std::move(_doc));
+
+    _doc.clear();
+}
```

Comment:
Just out of curiosity, why are we passing a pointer in the parser instead of just a value? That seems to be the only reason we need `clear` and `docClear` in the first place (I initially read this is _clear must be a nop since we moved from the value_ -- but it is actually a pointer 😬).

I'd probably prefer this taking a `DocString`, and the caller modifying its state.

Reply:
Fair. Let me revisit this.